opensearch_py_ml/ml_models/sentencetransformermodel.py

```diff
@@ -323,12 +314,11 @@ def read_queries(self, read_path: str, overwrite: bool = False) -> pd.DataFrame:
         passages = []
         for j in range(0, num_file):
             for dict_str in process[j]:
-                if "probability" in dict_str.keys():
-                    prob.append(dict_str["probability"])
-                if "passage" in dict_str.keys():
-                    passages.append(dict_str["passage"])
-                if "query" in dict_str.keys():
+                if "query" in dict_str.keys() and "passage" in dict_str.keys():
                     query.append(dict_str["query"])
+                    passages.append(dict_str["passage"])
+                    if "probability" in dict_str.keys():
+                        prob.append(dict_str["probability"])  # language modeling score
 
         df = pd.DataFrame(
```

Comment:
This will lead to wrong answers whenever len(prob) is different from len(query). At best zip will throw an error and at worst it will create the wrong dataset i.e queries and passages will get mismatched with prob. One solution could be to append "-1" whenever probability is not present in dict_str.keys(). Here "-1" will serve as a label saying that the probability does that exist.